src/lib.rs

```diff
@@ -1226,31 +1310,24 @@ impl<T> VecList<T> {
   }
 
   /// Convenience function for returning setting the offset head index.
-  fn set_head(&mut self, index: usize) {
-    self.head = Some(NonZeroUsize::new(index + 1).expect("head should not be 0"));
+  fn set_head(&mut self, index: NonMaxUsize) {
```

Comment:
set_head, set_tail, and set_vacant_head should probably just be inlined at this point instead of separate functions.